packages/core/schematics/migrations/standalone-false/migration.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import ts from 'typescript';
+import {ChangeTracker} from '../../utils/change_tracker';
+import {getImportSpecifier, getNamedImports} from '../../utils/typescript/imports';
+
+const CORE = '@angular/core';
+const DIRECTIVE = 'Directive';
+const COMPONENT = 'Component';
+const PIPE = 'Pipe';
+
+type RewriteFn = (startPos: number, width: number, text: string) => void;
+
+export function migrateFile(sourceFile: ts.SourceFile, rewriteFn: RewriteFn) {
+  const changeTracker = new ChangeTracker(ts.createPrinter());
+
+  // Check if there are any imports of the `AfterRenderPhase` enum.
+  const coreImports = getNamedImports(sourceFile, CORE);
+  if (!coreImports) {
+    return;
+  }
+  const directive = getImportSpecifier(sourceFile, CORE, DIRECTIVE);
+  const component = getImportSpecifier(sourceFile, CORE, COMPONENT);
+  const pipe = getImportSpecifier(sourceFile, CORE, PIPE);
+
+  if (!directive && !component && !pipe) {
+    return;
+  }
+
+  ts.forEachChild(sourceFile, function visit(node: ts.Node) {
+    ts.forEachChild(node, visit);
+
+    // First we need to check for class declarations
+    // Decorators will come after
+    if (!ts.isClassDeclaration(node)) {
+      return;
+    }
+
+    ts.getDecorators(node)?.forEach((decorator) => {
+      if (!ts.isDecorator(decorator)) {
+        return;
+      }
+
+      const callExpression = decorator.expression;
+      if (!ts.isCallExpression(callExpression)) {
+        return;
+      }
+
+      const decoratorIdentifier = callExpression.expression;
+      if (!ts.isIdentifier(decoratorIdentifier)) {
+        return;
+      }
+
+      // Checking the identifier of the decorator by comparing to the import specifier
+      switch (decoratorIdentifier.text) {
+        case directive?.name.text:
+        case component?.name.text:
+        case pipe?.name.text:
+          break;
+        default:
+          // It's not a decorator to migrate
+          return;
+      }
+
+      const [firstArg] = callExpression.arguments;
+      if (firstArg && !ts.isObjectLiteralExpression(firstArg)) {
```

Comment:
```suggestion
      if (!firstArg || !ts.isObjectLiteralExpression(firstArg)) {
```

This would otherwise crash for `@Directive()`

Reply:
btw, shoud we migrate `@Directive()` to `standalone:false` ?

Reply:
The answer is: no :)

Reply:
I'd suggest to still keep the test though, to have this case included in our test coverage